src/IndexManager.php

```diff
@@ -157,6 +163,17 @@ public function count($query, $className, array $parameters = [])
         return $this->engine->count($query, $this->getFullIndexName($className));
     }
 
+    protected function shouldBeIndexed($className, &$entity)
+    {
+        if ($methodName = $this->indexIfMapping[$className]) {
+            if (method_exists($entity, $methodName)) {
```

Comment:
why don't you use `PropertyAccess` ?  https://symfony.com/doc/current/components/property_access.html

It's more flexible, can read from Object and Arrays and is safe

Reply:
Why not but it depends what for. Did you mean to replace to `method_exists` call or do you see a way to rewrite the function in a more elegant way?

Reply:
```php
$propertyAccessor = PropertyAccess::createPropertyAccessor();

if ($propertyAccessor->isReadable($entity, $accessor) {
    return (bool) $propertyAccessor->getValue($entity, $accessor)
}
```

Also, with this, you should rename `methodName`  to `accessor` for better clarity in my opinion.

Reply:
In this case, I'm not sure what value this brings. Is there a case where `method_exists` will fail but the PropertyAccess will work?

Reply:
`method_exist` will return false in this case: 

```yml
  index_if: indexable
```

```php
protected $indexable = false;
// no getter or setter for $indexable.
```

when `$propertyAccessor->getValue($entity, 'indexable'))` does work

Also this won't work with `method_exists`:

```yml
  index_if: config.indexable
```

```php
protected $config = ['indexable' => false];
```

when `$propertyAccessor->getValue($entity, 'config.indexable'))` does work

--- 
well you're in Symfony context, using property accessor will bring more flexibility, I really don't see any disavantage of using a component of the ecosystem you're developing a bundle for.

Reply:
I didn't think about class properties, in this case, it does bring a lot a value! 🚀 I'll update the PR. Thanks for reviewing :)

Reply:
👍